.pytool/Plugin/LineEndingCheck/LineEndingCheck.py

```diff
@@ -0,0 +1,155 @@
+]
+
+ALLOWED_LINE_ENDING = b'\r\n'
+
+
+class LineEndingCheckBadLineEnding(Exception):
+    pass
+
+
+class LineEndingCheck(ICiBuildPlugin):
+    """
+    A CiBuildPlugin that checks whether line endings are a certain format.
+
+    By default, the plugin runs against all files in a package unless a
+    specific file or file extension is excluded.
+
+    Configuration options:
+    "LineEndingCheck": {
+        "IgnoreFiles": [],            # File patterns to ignore.
+    }
+    """
+
+    def GetTestName(self, packagename: str, environment: VarDict) -> Tuple:
+        """ Provide the testcase name and classname for use in reporting
+
+            Args:
+              packagename: string containing name of package to build
+              environment: The VarDict for the test to run in
+            Returns:
+                A tuple containing the testcase name and the classname
+                (testcasename, classname)
+                testclassname: a descriptive string for the testcase can
+                               include whitespace
+                classname: should be patterned <packagename>.<plugin>
+                           .<optionally any unique condition>
+        """
+        return ("Check line endings in " + packagename, packagename +
+                "." + PLUGIN_NAME)
+
+    def RunBuildPlugin(self, package_rel_path: str, edk2_path: Edk2Path,
+                       package_config: Dict[str, List[str]],
+                       environment_config: Any,
+                       plugin_manager: PluginManager,
+                       plugin_manager_helper: HelperFunctions,
+                       tc: JunitReportTestCase, output_stream=None) -> int:
+        """
+        External function of plugin. This function is used to perform the task
+        of the CiBuild Plugin.
+
+        Args:
+          - package_rel_path: edk2 workspace relative path to the package
+          - edk2_path: Edk2Path object with workspace and packages paths
+          - package_config: Dictionary with the package configuration
+          - environment_config: Environment configuration
+          - plugin_manager: Plugin Manager Instance
+          - plugin_manager_helper: Plugin Manager Helper Instance
+          - tc: JUnit test case
+          - output_stream: The StringIO output stream from this plugin
+            (logging)
+
+        Returns
+          >0 : Number of errors found
+          0  : Ran successfully
+          -1 : Skipped due to a missing pre-requisite
+        """
+
+        abs_pkg_path = \
+            edk2_path.GetAbsolutePathOnThisSystemFromEdk2RelativePath(
+                        package_rel_path)
+
+        if abs_pkg_path is None:
+            tc.SetSkipped()
+            tc.LogStdError(f"Package folder not found {abs_pkg_path}")
+            return 0
+
+        all_files = [n for n in glob.glob(os.path.join(abs_pkg_path, '*.*'),
+                     recursive=True)]
+
+        if "IgnoreFiles" in package_config:

```

Comment:
why not leverage the more capable gitignore syntax?

Reply:
Updated to use gitignore syntax